plugins/connection/nsenter.py

```diff
@@ -0,0 +1,246 @@
+# (c) 2021 Jeff Goldschrafe <jeff@holyhandgrenade.org>
+# Based on Ansible local connection plugin by:
+# (c) 2012 Michael DeHaan <michael.dehaan@gmail.com>
+# (c) 2015, 2017 Toshio Kuratomi <tkuratomi@ansible.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: nsenter
+    short_description: execute on host running controller container
+    version_added: 1.9.0
+    description:
+        - This connection plugin allows Ansible, running in a privileged container, to execute tasks on the container
+          host instead of in the container itself.
+        - This is useful for running Ansible in a pull model, while still keeping the Ansible control node
+          containerized.
+        - It relies on having privileged access to run nsenter in the host's PID namespace, allowing it to enter the
+          namespaces of the provided PID (default PID 1, or init/systemd).
+    author: Jeff Goldschrafe (@jgoldschrafe)
+    options:
+        nsenter_pid:
+            description:
+                - PID to attach with using nsenter.
+                - The default should be fine unless you're attaching as a non-root user.
+            type: int
+            default: 1
+            vars:
+                - name: ansible_nsenter_pid
+            env:
+                - name: ANSIBLE_NSENTER_PID
+            ini:
+                - section: nsenter_connection
+                  key: nsenter_pid
+        pipelining:
+            default: ANSIBLE_PIPELINING
```

Comment:
That's a string, not a boolean. The default must be `true` or `false`.

(Personally I would remove this whole option and do what the `local` plugin has been doing until ansible-core 2.11, namely ignore it.)

Reply:
if it were removed I might recommend keeping at least part of it (the name?) around, and commented out, with a note about why, just so that if the parts from `local` are "refreshed" it doesn't get brought back in accidentally.

Reply:
For that a remark in line https://github.com/ansible-collections/community.docker/pull/173/files#diff-fd1d8cb72d9ad37889af492d0e7412b05cb16311b91327493e556af76b7ba645R137 should be enough.